Source/WebCore/editing/Editor.cpp

```diff
@@ -3691,6 +3696,13 @@ void Editor::removeTextPlaceholder(TextPlaceholderElement& placeholder)
     // To match the Legacy WebKit implementation, set the text insertion point to be before where the placeholder used to be.
     if (document->selection().isFocusedAndActive() && document->focusedElement() == savedRootEditableElement)
         document->selection().setSelection(VisibleSelection { savedPositionBeforePlaceholder }, FrameSelection::defaultSetSelectionOptions(UserTriggered::Yes));
+
+    auto caretVisibilitySuppressionReasons = document->selection().caretVisibilitySuppressionReasons();
+    caretVisibilitySuppressionReasons.remove(CaretVisibilitySuppressionReason::TextPlaceholderIsShowing);
+
+    auto caretVisibility = !caretVisibilitySuppressionReasons.contains(CaretVisibilitySuppressionReason::IsFocusedOrActive) ? CaretVisibility::Visible : CaretVisibility::Hidden;
```

Comment:
Ideally you'd have a centralized place that keys off of existing reasons and computes a visibility. You shouldn't need to check for one reason in an unrelated part of the code.

Reply:
fixed